docs/sc-memory/api/cpp/extended/agents/actions.md

```diff
@@ -0,0 +1,345 @@
+...
+```
+
+### **GetClass**
+
+To get class of action use this method.
+
+```cpp
+...
+ScAddr const & actionClassAddr = action.GetClass();
+...
+```
+
+### **GetArgument**
+
+Actions can have arguments. Action arguments are some objects on which this action should be performed. If action has arguments in knowledge base, then you will get them. There are a couple of getters for this.
+
+```cpp
+...
+// Provide relation here if your argument has specific role in action. 
+ScAddr const & argAddr = action.GetArgument(ScKeynodes::rrel_1);
+// If there is no argument with such role, then `argAddr` will be empty.
+...
+```
+
+```cpp
+...
+// Or provide index of argument. 1 is equal to `ScKeynodes::rrel_1`,
+// 2 is equal to `ScKeynodes::rrel_2` and etc.
+ScAddr const & argAddr = action.GetArgument(1);
+// If there is no argument with such role, then `argAddr` will be empty.
+...
+```
+
+!!! warning
+    This method will throw `utils::ExceptionInvalidParams` if you provide invalid index for argument (for example, 0).
+
+```cpp
+...
+ScAddr const & argAddr = action.GetArgument(1, defaultArgumentAddr);
+// If there is no argument with such role, then `argAddr` will equal 
+// to `defaultArgumentAddr`.
+...
+```
+
+### **GetArguments**
+
+You can get all arguments using one getters.
+
+```cpp
+...
+// Here, 2 is number arguments in specified action.
+auto const & [argAddr1, argAddr2] = action.GetArguments<2>();
+// If action doesn't have some argument, it will be empty.
+...
+```
+
+```cpp
+...
+// You can't ignore some arguments.
+auto const & [argAddr1, argAddr2, _] = action.GetArguments<3>();
+...
+```
+
+!!! note
+    `GetArguments` uses `GetArgument` for each argument.
+
+### **SetArgument**
+
+You can set arguments for specified action.
+
+```cpp
+...
+// Provide relation here if your argument should have specific role in action. 
+action.SetArgument(ScKeynodes::rrel_1, argAddr);
+...
+```
+
+```cpp
+...
+// Or provide index of argument. 1 is equal to `ScKeynodes::rrel_1`,
+// 2 is equal to `ScKeynodes::rrel_2` and etc.
+action.SetArgument(1, argAddr);
+...
+```
+
+!!! note
+    If action has already argument with specified role, then connection between action and this argument will be removed and created new one between action and new argument.
+
+### **SetArguments**
+
+```cpp
+...
+// Set several arguments simultaneously.
+action.SetArguments(argAddr1, argAddr2);
+...
+```
+
+!!! note
+    `SetArguments` uses `SetArgument` for each argument.
+
+### **Action result**
+
+All action should have result (result situation). Result situation is structure that contains all sc-constructions that which indicate result of interpreting action.
+
+#### **GetResult**
+
+You can result for any actions.
+
+```cpp
+...
+// Use this method after that some agent finished 
+// interpreting action.
+ScStructure const & actionResult = action.GetResult();
+...
+```
+
+!!! warning
+    You can call this method for finished action only.
```

Comment:
why can't I call this method for not finished action?

Reply:
It prevents a situation where an agent that performs an action is still forming a result for that action, and you haven't waited for it and already want to get result for that action.

Reply:
I mean I can call it, just an exception will be thrown

Reply:
Fixed